compiler/record-minmax/src/RecordMinMax.cpp

```diff
@@ -160,9 +145,17 @@ void RecordMinMax::profileData(const std::string &input_data_path)
     auto node = iter->first;
     auto minmax = iter->second;
 
-    float min = getMovingAverage(minmax.min_vector, 0.9);
-    float max = getMovingAverage(minmax.max_vector, 0.9);
-
+    float min, max;
+    if (mode == "percentile")
+    {
+      min = getNthPercentile(minmax.min_vector, min_percentile);
+      max = getNthPercentile(minmax.max_vector, max_percentile);
+    }
+    else if (mode == "moving_average")
+    {
+      min = getMovingAverage(minmax.min_vector, 0.9);
+      max = getMovingAverage(minmax.max_vector, 0.9);
+    }
```

Comment:
if the `mode` is not of the two, `min`/`max` will be undefined. Is the a check code anywhere for `mode` ?

Reply:
I've added a check code. Thanks.